processor/batchprocessor/batch_processor.go

```diff
@@ -327,3 +336,61 @@ func (bm *batchMetrics) add(pm pdata.Metrics) {
 	bm.itemCount = bm.itemCount + uint32(md.MetricCount())
 	md.ResourceMetrics().MoveAndAppendTo(bm.metricData.ResourceMetrics())
 }
+
+// splitTraces splits traces into one item of size1 and remaining items of size2.
+func splitTraces(size1, size2 uint32, toSplit pdata.Traces) []pdata.Traces {
+	if toSplit.SpanCount() <= int(size1) {
+		return []pdata.Traces{toSplit}
+	}
+	var splitTraces []pdata.Traces
+	splitTraces = append(splitTraces, splitTrace(size1, toSplit))
+	for toSplit.SpanCount() > int(size2) {
+		splitTraces = append(splitTraces, splitTrace(size2, toSplit))
+	}
+	if toSplit.SpanCount() > 0 {
+		splitTraces = append(splitTraces, toSplit)
+	}
+	return splitTraces
+}
+
+// splitTrace removes spans from the input trace and returns a new trace of the specified size.
+func splitTrace(size uint32, toSplit pdata.Traces) pdata.Traces {
+	if toSplit.SpanCount() <= int(size) {
+		return toSplit
+	}
+	copied := uint32(0)
+	dest := pdata.NewTraces()
+	rss := toSplit.ResourceSpans()
+	for i := rss.Len() - 1; i >= 0; i-- {
+		rs := rss.At(i)
+		copiedRs := pdata.NewResourceSpans()
+		copiedRs.InitEmpty()
+		rs.Resource().CopyTo(copiedRs.Resource())
+		dest.ResourceSpans().Append(&copiedRs)
+
+		for j := rs.InstrumentationLibrarySpans().Len() - 1; j >= 0; j-- {
+			instSpans := rs.InstrumentationLibrarySpans().At(j)
+			copiedInstSpans := pdata.NewInstrumentationLibrarySpans()
+			copiedInstSpans.InitEmpty()
+			instSpans.InstrumentationLibrary().CopyTo(copiedInstSpans.InstrumentationLibrary())
+			copiedRs.InstrumentationLibrarySpans().Append(&copiedInstSpans)
+			for k := instSpans.Spans().Len() - 1; k >= 0 && copied < size; k-- {
+				span := instSpans.Spans().At(k)
+				copiedInstSpans.Spans().Append(&span)
```

Comment:
This might not be ideal in terms of resizing the array. The size could be initialized before the loop.

Reply:
fixed the size is set before the loop